core/deployment/src/main/java/io/quarkus/artemis/core/deployment/DevServicesArtemisProcessor.java

```diff
@@ -193,17 +193,21 @@ private static RunningDevService startArtemis(
             DockerStatusBuildItem dockerStatusBuildItem,
             ArtemisDevServiceCfg config,
             LaunchModeBuildItem launchMode,
-            Optional<Duration> timeout) {
+            @SuppressWarnings("OptionalUsedAsFieldOrParameterType") Optional<Duration> timeout) {
```

Comment:
Can we not set this at the class level, it is how Quarkus processors work.

Reply:
Done. For the extension part, I have moved this suppress up to all class-levels in all extension-modules.